spec/rspec/mocks/reraising_eager_raises_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "spec_helper"
+
+RSpec.describe "Reraising eager raises during the verify step" do
+  it "does not reraise when a double receives a message that hasn't been allowed/expected" do
+    with_unfulfilled_double do |dbl|
+      expect { dbl.foo }.to fail
+      expect { verify_all }.not_to raise_error
+    end
+  end
+
+  it "reraises when a negative expectation receives a call" do
+    with_unfulfilled_double do |dbl|
+      expect(dbl).not_to receive(:foo)
+      expect { dbl.foo }.to fail
+      expect { verify_all }.to raise_error(/expected: 0 times with any arguments/)
+    end
+  end
+
+  it "reraises when an expectation with a count is violated" do
```

Comment:
I think "exceeded" would be better than "violated".  "violated" could refer to either too few invocations or too many but this really only applies to when there are too many, and "exceeded" is a better word for that.